app/javascript/components/generic-objects-form/index.jsx

```diff
@@ -45,10 +45,26 @@ const GenericObjectForm = ({ recordId }) => {
           // check to display file upload/edit component depending on whether the generic object being edited already has a custom image or not
           initialValues.image_update = !initialValues.picture;
           delete initialValues.properties;
-          setState({ initialValues, isLoading: false });
+          setState({
+            initialValues,
+            isLoading: false,
+            classOptions: Object.keys(allowed_association_types).map((key) => ({
```

Comment:
We can move this to a common method as this repeats in line 62 and make it fall in one line..